### ISSUE TITLE
Prefetch one more word

### ISSUE THREAD

**Issue by larsbrinkhoff** [nocrew/ostis#141]
More completely model the IRC/IR/IRD prefetch queue.

**Comment by stefanberndtsson** [comment]
Can this be done in a way compatible with the old cpu code?

**Comment by larsbrinkhoff** [comment]
Yes, I think so.  I started doing that a week ago, but ran into a dead end.

But there's a great advantage in having the old instructions work more correctly, because they're mixed with the new instructions.

**Comment by larsbrinkhoff** [comment]
Basically, all instructions that do a `bus_read_<byte/word/long>` from `cpu->pc`, have to be changed to use `fetch_instr` instead.

Also, there's some trickery going on with `cpu->pc` being incremented or not.  I'm guessing that the old `ea_read` functions can be part of a read-modify-write operation, in which case the PC should be left untouched until the `ea_write` is done?

**Comment by larsbrinkhoff** [comment]
One possible shortcut would be to have all old instructions clear out the prefetch queue at the beginning, and fill it at the end according to the value of `cpu->pc`.

This quick-and-dirty hack would only be to provide an execution environment for the new clocked instructions.

**Comment by larsbrinkhoff** [comment]
By the way, when I wrote a VAXstation 100 emulator, I wrote a 68000 engine that models the prefetch queue.  https://github.com/larsbrinkhoff/absolutely-not-a-vaxstation100-emulator/
